File: CS/WindowsApplication103/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Utils;
using DevExpress.XtraBars;
using System.Reflection;
using System.Collections;

namespace WindowsApplication103
{
    public partial class Form1 : Form, IMessageFilter
    {
        public Form1()
        {
            InitializeComponent();
            Application.AddMessageFilter(this);
            ToolTipList.Add(new ToolTipObject(barButtonItem1,"ToolTip","CToolTip"));
            ToolTipList.Add(new ToolTipObject(barButtonItem2,"AnotherToolTip","ToolTipShownInCustomizationMode"));
            ToolTipList.Add(new ToolTipObject(barButtonItem3, "Some","It's Work!"));

        }

        List<ToolTipObject> ToolTipList = new List<ToolTipObject>();
        const int WM_MouseMove = 0x0200;


        private Rectangle GetLinksScreenRect(BarItemLink link)
        {
            PropertyInfo info = typeof(BarItemLink).GetProperty("BarControl", BindingFlags.Instance | BindingFlags.NonPublic);
            Control c = (Control)info.GetValue(link, null);
            return c.RectangleToScreen(link.Bounds);
        }



        #region IMessageFilter Members

        public bool PreFilterMessage(ref Message m)
        {
                Rectangle BarItemRect = new Rectangle();
                int index = 0;
                if (m.Msg == WM_MouseMove)
                {
                    for (int i = 0; i < ToolTipList.Count; i++)
                    {
                        try
                        {
                            BarItemRect = GetLinksScreenRect(ToolTipList[i].ToolTipItem.Links[0]);
                        }
                        catch { }
                        if (!BarItemRect.IsEmpty)
                            if (BarItemRect.Contains(MousePosition)) { index = i; break; }

                    }

                    if (BarItemRect.Contains(MousePosition))
                    {
                        ToolTipControllerShowEventArgs te = new ToolTipControllerShowEventArgs();
                        te.ToolTipLocation = ToolTipLocation.Fixed;
                        te.SuperTip = new SuperToolTip();
                        te.SuperTip.Items.Add(!barManager1.IsCustomizing? ToolTipList[index].ToolTip : ToolTipList[index].ToolTipCustomizationMode);
                        Point linkPoint = new Point(BarItemRect.Right, BarItemRect.Bottom);
                        barManager1.GetToolTipController().ShowHint(te, linkPoint);
                    }
                    else barManager1.GetToolTipController().HideHint();

                }
            
            return false;
        }

        #endregion
    }
}

File: VB/WindowsApplication103/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Utils
Imports DevExpress.XtraBars
Imports System.Reflection
Imports System.Collections

Namespace WindowsApplication103
	Partial Public Class Form1
		Inherits Form
		Implements IMessageFilter
		Public Sub New()
			InitializeComponent()
			Application.AddMessageFilter(Me)
			ToolTipList.Add(New ToolTipObject(barButtonItem1,"ToolTip","CToolTip"))
			ToolTipList.Add(New ToolTipObject(barButtonItem2,"AnotherToolTip","ToolTipShownInCustomizationMode"))
			ToolTipList.Add(New ToolTipObject(barButtonItem3, "Some","It's Work!"))

		End Sub

		Private ToolTipList As New List(Of ToolTipObject)()
		Private Const WM_MouseMove As Integer = &H0200


		Private Function GetLinksScreenRect(ByVal link As BarItemLink) As Rectangle
			Dim info As PropertyInfo = GetType(BarItemLink).GetProperty("BarControl", BindingFlags.Instance Or BindingFlags.NonPublic)
			Dim c As Control = CType(info.GetValue(link, Nothing), Control)
			Return c.RectangleToScreen(link.Bounds)
		End Function



		#Region "IMessageFilter Members"

		Public Function PreFilterMessage(ByRef m As Message) As Boolean Implements IMessageFilter.PreFilterMessage
				Dim BarItemRect As New Rectangle()
				Dim index As Integer = 0
				If m.Msg = WM_MouseMove Then
					For i As Integer = 0 To ToolTipList.Count - 1
						Try
							BarItemRect = GetLinksScreenRect(ToolTipList(i).ToolTipItem.Links(0))
						Catch
						End Try
						If (Not BarItemRect.IsEmpty) Then
							If BarItemRect.Contains(MousePosition) Then
								index = i
								Exit For
							End If
						End If

					Next i

					If BarItemRect.Contains(MousePosition) Then
						Dim te As New ToolTipControllerShowEventArgs()
						te.ToolTipLocation = ToolTipLocation.Fixed
						te.SuperTip = New SuperToolTip()
						If (Not barManager1.IsCustomizing) Then
							te.SuperTip.Items.Add(ToolTipList(index).ToolTip)
						Else
							te.SuperTip.Items.Add(ToolTipList(index).ToolTipCustomizationMode)
						End If
						Dim linkPoint As New Point(BarItemRect.Right, BarItemRect.Bottom)
						barManager1.GetToolTipController().ShowHint(te, linkPoint)
					Else
						barManager1.GetToolTipController().HideHint()
					End If

				End If

			Return False
		End Function

		#End Region
	End Class
End Namespace